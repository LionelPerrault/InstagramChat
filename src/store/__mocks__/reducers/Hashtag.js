import posts from './posts'

export const hashtags = [
  { hashtag: '#travel' },
  { hashtag: '#cool' },
  { hashtag: '#hmm' },
]

export default {
  popularHashtags: hashtags,
  groupHashtags: hashtags,
  userHashtags: hashtags,
  hashtagPosts: posts,
}
