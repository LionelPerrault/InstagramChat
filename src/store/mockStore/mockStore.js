import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
