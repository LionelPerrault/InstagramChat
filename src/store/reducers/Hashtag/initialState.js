export default {
  userHashtags: [],
  groupHashtags: [],
  popularHashtags: [],
  hashtagPosts: [],
}
