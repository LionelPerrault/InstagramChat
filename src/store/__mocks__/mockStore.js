import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
