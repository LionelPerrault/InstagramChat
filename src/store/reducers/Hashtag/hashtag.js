/* eslint indent:0 */
/* eslint no-unreachable:0 */

import initialState from './initialState'

export default (state = initialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_USER_HASHTAGS':
      return { ...state, userHashtags: py }
      break

    case 'GET_GROUP_HASHTAGS':
      return { ...state, groupHashtags: py }
      break

    case 'GET_POPULAR_HASHTAGS':
      return { ...state, popularHashtags: py }
      break

    case 'GET_HASHTAG_POSTS':
      return { ...state, hashtagPosts: py }
      break
  }

  return state
}
