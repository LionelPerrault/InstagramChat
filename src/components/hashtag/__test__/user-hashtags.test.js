import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../store/__mocks__/mockStore'
import UserHashtags from '../user-hashtags'
import Hashtag from '../../../store/__mocks__/reducers/Hashtag'

describe('UserHashtags Component', () => {
  const comp = (
    <Provider store={mockStore}>
      <UserHashtags username="takkar" />
    </Provider>
  )

  it('should match snapshot', () => {
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with null when hashtags.length == 0', () => {
    Hashtag.userHashtags = []
    const tree = create(comp).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
