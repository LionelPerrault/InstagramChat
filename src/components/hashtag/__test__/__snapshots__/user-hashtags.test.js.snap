// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHashtags Component should match snapshot 1`] = `
<div>
  <div
    className="recomm user-hashtags"
  >
    <div
      className="recomm_top"
    >
      <span>
        Your recent hashtags
      </span>
    </div>
    <div
      className="recomm_main"
    >
      <Link
        className="uh_link"
        to="/hashtag/travel"
      >
        #travel
      </Link>
      <Link
        className="uh_link"
        to="/hashtag/cool"
      >
        #cool
      </Link>
      <Link
        className="uh_link"
        to="/hashtag/hmm"
      >
        #hmm
      </Link>
    </div>
  </div>
</div>
`;

exports[`UserHashtags Component should match snapshot with null when hashtags.length == 0 1`] = `<div />`;
